source/common/upstream/original_dst_cluster.cc

```diff
@@ -116,7 +116,7 @@ OriginalDstCluster::OriginalDstCluster(
                            : false),
       host_map_(std::make_shared<HostMap>()) {
   // TODO(dio): Remove hosts check once the hosts field is removed.
-  if (config.has_load_assignment() || !config.hosts().empty()) {
```

Comment:
I thought that the point of doing conversion was to not have to deal with the old style things in Envoy
Is this due to the special case TODO above?

Reply:
So, in this case, we don't do any conversion from hosts --> load assignment, since neither of these are used by original dst clusters. So, here we just check that both are not set.